src/world/partner/bombette.c

```diff
@@ -119,14 +138,139 @@ s32 func_802BD748(void) {
+}
+
+s32 func_802BE520_319270(Npc* npc1, Npc* npc2) {
+    f32 sp20, sp24, sp28;
+    f32 temp_f6, temp_f20, temp_f22, temp_f24, temp_f26, temp_f28;
+    s32 slippingResult;
+    f32 distance;
+    s32 ret;
+    f32 temp2;
+    f32 height;
+
+    if (D_802BE928 == 0) {
+        return FALSE;
+    }
+
+    temp_f26 = npc2->pos.x;
+    temp_f28 = npc1->pos.x;
+
+    sp20 = npc2->pos.x - npc1->pos.x;
+    sp24 = (npc2->pos.y + (f32)(npc2->collisionHeight * 0.5) - npc1->pos.y);
+    temp_f22 = npc2->pos.z;
+    temp_f24 = npc1->pos.z;
+    sp28 = temp_f22 - temp_f24;
+    temp_f6 = npc2->collisionRadius * 0.55;
+    height = 35.0f;
+
+    temp2 = sqrtf(SQ(sp20) + SQ(sp24) + SQ(sp28));
+    ret = FALSE;
+
+    if (temp2 < (temp_f6 + height)) {
+        ret = TRUE;
+    }
+
+    temp_f20 = atan2(temp_f26, temp_f22, temp_f28, temp_f24);
+    distance = dist2D(temp_f26, temp_f22, temp_f28, temp_f24);
+
+    sp20 = npc1->pos.x;
+    sp24 = npc1->pos.y;
+    sp28 = npc1->pos.z;
+
+    slippingResult = npc_test_move_taller_with_slipping(0, &sp20, &sp24, &sp28, distance, temp_f20, height, 2.0f);
+
+    if (slippingResult) {
+        return FALSE;
     }
-    return partner_put_away(unk) != 0;
+
+    return ret;
 }
 
-INCLUDE_ASM(s32, "world/partner/bombette", func_802BE520_319270);
+void func_802BE6E8_319438(Npc* npc) {
+    PlayerStatus* playerStatus = &gPlayerStatus;
+    PartnerActionStatus* partnerActionStatus = &gPartnerActionStatus;
+    f32 phi_f12;
+    f32* new_var0;
+    f32* new_var1;
+
+    if (partnerActionStatus->actionState.b[0] != 0) {
+        if (D_802BE92C) {
+            enable_player_input();
+        }
+
+        D_802BE928 = 0;
+        playerStatus->flags &= -3;
+        npc->jumpVelocity = 0.0f;
+        npc->flags &= ~0x800;
+
+        set_action_state(0);
+        partner_clear_player_tracking(npc);
+
+        partnerActionStatus->actionState.b[0] = 0;
+        partnerActionStatus->actionState.b[3] = 0;
+
+        npc->pos.x = playerStatus->position.x;
+        npc->pos.y = playerStatus->position.y;
+        npc->pos.z = playerStatus->position.z;
+
+        if (D_802BE920 == 0) {
+            add_vec2D_polar(&npc->pos.x, &npc->pos.z, playerStatus->colliderDiameter / 4, clamp_angle(playerStatus->targetYaw + 90.0f));
+        } else {
+            add_vec2D_polar(&npc->pos.x, &npc->pos.z, playerStatus->colliderDiameter / 4, clamp_angle(playerStatus->targetYaw - 90.0f));
+        }
+
+        npc->jumpVelocity = 0.0f;
+        npc->pos.y = playerStatus->position.y;
+        npc->rotation.x = 0.0f;
+        npc->rotation.z = 0.0f;
+        npc->currentAnim.w = 0x30003;
+        partner_clear_player_tracking(npc);
+        disable_npc_blur(npc);
+
+        if (D_802BE924) {
+            D_802BE924 = 0;
+            sfx_stop_sound(0x80000000);
+        }
+    }
+}
+
+EvtSource D_802BE880_3195D0 = {
+    EVT_CALL(func_802BD300_318050)
+    EVT_RETURN
+    EVT_END
+};
+
+unkPartnerStruct* D_802BE89C_3195EC = (unkPartnerStruct*)0x802BE940;
```

Comment:
Is there no symbol for this? Should probably be added to undefined_syms if not rather than adding a raw pointer. Otherwise, `// TODO raw pointer` this